api/src/main/java/org/openmrs/module/reporting/common/SqlIterator.java

```diff
@@ -0,0 +1,90 @@
+package org.openmrs.module.reporting.common;
+
+import org.openmrs.module.reporting.dataset.DataSetColumn;
+import org.openmrs.module.reporting.dataset.DataSetRow;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+public class SqlIterator implements Iterator<DataSetRow> {
+    private ResultSet resultSet;
+    private List<DataSetColumn> columns;
+    private Statement statement;
+    private Connection connection;
+
+    public SqlIterator(ResultSetMetaData metadata, ResultSet resultSet, Connection connection,Statement statement) throws SQLException {
+        this.resultSet = resultSet;
+        this.connection = connection;
+        this.statement = statement;
+        this.createDataSetColumns(metadata);
+    }
+
+    public boolean hasNext() {
+        try {
+            if(resultSet.isLast()){
+                closeConnection();
+                return false;
+            }else{
+                return true;
+            }
+        } catch (SQLException throwables) {
+            return false;
```

Comment:
Are you intentionally ignoring this exception?

Reply:
I probably have there because and old development where I was closing the connection there.
But you are right I should do something, not sure what. I cannot throw one exception here since this is an Override.
Maybe log and return false?

Reply:
Yes logging and returning false looks better than just ignoring. Isn't `last result` exposing implementation details to the client?